src/ert/scheduler/local_driver.py

```diff
@@ -2,19 +2,62 @@
 
 import asyncio
 import os
-from typing import MutableMapping
+from asyncio.subprocess import Process
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Generic,
+    MutableMapping,
+    Optional,
+    Protocol,
+    TypeVar,
+)
 
 from ert.scheduler.driver import Driver, JobEvent
 
+_T = TypeVar("_T")
+_T_co = TypeVar("_T_co", covariant=True)
+_T_contra = TypeVar("_T_contra", contravariant=True)
 
-class LocalDriver(Driver):
-    def __init__(self) -> None:
+
+class _InitFn(Protocol, Generic[_T_co]):
+    async def __call__(
+        self, iens: int, executable: str, /, *args: str, cwd: str
+    ) -> _T_co:
+        ...
+
+
+_WaitFn = Callable[[_T_contra], Awaitable[bool]]
+_KillFn = Callable[[_T_contra], Awaitable[None]]
+
+
+class TaskDriver(Driver, Generic[_T]):
```

Comment:
I'm bit lost what's this `_T` template. Can it be anything else than `asyncio.subprocess.Process`? Regardless the driver we use we need to execute bsub via the `Process`? Or is it about hiding the polling mechanism in a Process wrapper?

Reply:
It can in the case of `MockDriver`, where it is `None`. The `TaskDriver` has no polling. LFS and co will likely inherit directly from `Driver`.